mobile-app/app/screens/AppNavigator/screens/Dex/CompositeSwap/components/SlippageTolerance.tsx

```diff
@@ -1,124 +1,211 @@
+  const percentageList = ['1', '3', '5', '10', '20']
+  const [selectedSlippage, setSelectedSlippage] = useState(slippage.toString())
+  const [isRiskWarningDisplayed, setIsRiskWarningDisplayed] = useState(false)
+  const submitSlippage = debounce(onSubmitSlippage, 500)
+
+  const onSlippageChange = (value: string): void => {
+    setSelectedSlippage(value)
+    submitSlippage(new BigNumber(value), isCustomSlippage)
+  }
+
+  const isSlippageValid = (): boolean => {
+    return slippageError === undefined || slippageError?.type === 'helper'
+  }
+
+  const validateSlippage = (value: string): void => {
+    if (value === undefined || value === '') {
+      setSlippageError({
+        type: 'error',
+        text: translate('screens/SlippageTolerance', 'Required field is missing')
+      })
+      return
+    } else if (!(new BigNumber(value).gte(0) && new BigNumber(value).lte(100))) {
+      setSlippageError({
+        type: 'error',
+        text: translate('screens/SlippageTolerance', 'Slippage rate must range from 0-100%')
+      })
+      return
+    }
+
+    setSlippageError(undefined)
+  }
+
+  useEffect(() => {
+    validateSlippage(selectedSlippage)
+    setIsRiskWarningDisplayed((new BigNumber(selectedSlippage).gte(20) && new BigNumber(selectedSlippage).lte(100)))
+  }, [selectedSlippage])
+
   return (
-    <View
-      style={tailwind('flex-row')}
-    >
-      <View style={tailwind('w-6/12 flex-grow')}>
+    <>
+      <View style={tailwind('flex-row mb-3 flex-wrap')}>
+        {percentageList.map((value) => (
+          <SlippageButton
+            key={value}
+            onPress={() => {
+              setIsCustomSlippage(false)
+              onSlippageChange(value)
+            }}
+            isActive={!isCustomSlippage && selectedSlippage === value}
+            label={`${value}%`}
+          />
+        ))}
+        <SlippageButton
+          onPress={() => {
+            setIsCustomSlippage(true)
+          }}
+          icon={
+            <ThemedIcon
+              size={14}
+              name='edit'
+              iconType='MaterialIcons'
+              style={tailwind('mr-1')}
+              dark={tailwind(`${isCustomSlippage ? 'text-darkprimary-500' : 'text-gray-50'}`)}
+              light={tailwind(`${isCustomSlippage ? 'text-primary-500' : 'text-gray-900'}`)}
+            />
+          }
+          isActive={isCustomSlippage}
+          label={translate('screens/SlippageTolerance', 'Custom')}
+        />
+      </View>
+
+      {isCustomSlippage && (
+        <ThemedView
+          dark={tailwind('bg-transparent')}
+          light={tailwind('bg-transparent')}
+          style={tailwind('flex-row w-full')}
+        >
+          <WalletTextInput
+            autoFocus
+            onChangeText={onSlippageChange}
+            keyboardType='numeric'
+            autoCapitalize='none'
+            placeholder='0.00%'
+            style={tailwind('flex-grow h-8')}
```

Comment:
Let's add a width here so it doesn't overflow
![IMG_FA0F798D9CEB-1](https://user-images.githubusercontent.com/40191153/154419599-ff4843f2-69d8-460f-a7c3-370abca38f30.jpeg)